### ISSUE TITLE
Sass not loading from Vue component when compiling with Laravel Mix

### ISSUE THREAD

**Issue by dima2306** [JeffreyWay/laravel-mix#2819]
-   Laravel Mix Version: 6.0.11
-   Node Version (`node -v`): 14.15.4
-   NPM Version (`npm -v`): 6.14.9
-   webpack: 5.19.0
-   webpack-cli: 4.4.0
-   webpack-dev-server: 4.0.0-beta.0
-   OS: Fedora 33

### Description:
Hi everyone,

Upgraded to Laravel Mix 6 & Webpack 5 and have a problem with compiling. It finishes successfully but not including styles imported from withing Vue component. Specifically, I'm using [Vue Splide](https://github.com/Splidejs/vue-splide) adapter to display the slider on my main page and styling is broken.

I'm including styles like this:
```javascript
// Slider component...
<style lang="scss">
    @import '~@splidejs/splide/dist/css/themes/splide-default.min.css';
</style>
```

```javascript
// webpack.mix.js
mix.sass('resources/sass/app.scss', 'public/front/css')
   .js('resources/js/app.js', '')
   .vue()
   .alias({
     ziggy: path.resolve('vendor/tightenco/ziggy/src/js/route.js'),
     '@components': path.resolve(__dirname, 'resources/js/Components'),
     '@publicDir': path.resolve(__dirname, 'public')
   })
   .webpackConfig(config)
   .mergeManifest()
   .options({
     extractVueStyles: true,
   })
   .extract();
```

I tried without .options() also & still styles not included. But if I move `@import '~@splidejs/splide/dist/css/themes/splide-default.min.css';` to `app.scss` slider styles are included.

I only need slider styles within the Slider component so it doesn't make sense for move styles to `app.scss`.

This thing was working on Mixv5/WP4.

Am I missing something?

P.S.
I've also modified `.vue()` with:
```javascript
.vue({
    extractStyles: true,
})
```
But in this case, app tries to load `app.js` with `app.js?id=[chunkhash]` which is equal to `front0.js?id=[chunkhash]` and I'm getting error loading chunk file.

I do see that `app.css` file is slightly bigger in this case, which means that slider styles are included there from vue component `<style>`

Original post: https://laracasts.com/discuss/channels/elixir/sass-not-loading-from-vue-component-when-compiling-with-laravel-mix

### Steps To Reproduce:
Include sass/scss style from the Vue component and see if  it's compiled.

**Comment by roitto** [comment]
This issue might be associated with the issue #2589 ?

**Comment by dima2306** [comment]
Yes, @roitto I think my issue might be related to #2589.

Soo, I've added
```javascript
.vue({
   extractStyles: 'public/front/css/vendor.css',
});
```
And then I added in my template's head section
```html
<link href="{{ mix('front/css/vendor.css') }}" rel="stylesheet"></link>
```
The result is the same, the slider is not styled.

I also tried
```javascript
.vue({
   extractStyles: 'public/front/css/app.css',
});
```
And the result is the same error I described in the original post. I assume it's doing `extractStyles: true`. My app.css is definitely bigger but there are some errors with chunks then...

As I see they're working on this issue, hope it will be solved soon.

**Comment by beaubus** [comment]
looks like globalStyles not working too

```js
    mix.js('resources/js/backend.js', 'public/js')
       .vue({
            globalStyles: { sass: ['resources/sass/shared/settings/mixins.sass'] },
            runtimeOnly: false,
       });

```

"laravel-mix": "^6.0.11",
"sass-resources-loader": "^2.1.1",

**Comment by bhaidar** [comment]
Hello @JeffreyWay It seems loading global styles in a laravel/vue app is still not working. I am using v6.0.19. Thanks

**Comment by bhaidar** [comment]
Hey @beaubus Did you figure out how to fix the globalStyles issue?

**Comment by thecrypticace** [comment]
Hmm I'll take a look later this week. That should be working fine.

**Comment by beaubus** [comment]
> Hey @beaubus Did you figure out how to fix the globalStyles issue?

No, it was faster for me at that moment to just convert globalStyles to local styles.

**Comment by dima2306** [comment]
Does anybody know if the issue with globalStyles is resolved?

**Comment by GreenIPO** [comment]
In my case, not laravel mix problem. This helps me slove the problem
[Style in vue files is ignored when sideEffects: false is defined in package.json #718](https://github.com/vuejs/vuepress/issues/718#issuecomment-412345448)

**Comment by dima2306** [comment]
@GreenIPO Interesting, if the problem still exists it's definitely worth checking.

**Comment by mdsaifurrahmann** [comment]
Can you please try removing the
`extractVueStyles: true,` from `.options` ?
Also removing `extractStyles: true,` from inside of `.vue({})` and keep only the `.vue()`

in my case, this worked.

**Comment by kranthiboga23** [comment]
@mdsaifurrahmann  Yes is working for me as well by removing `extractStyles/extractVueStyles `from inside of .`vue({})` however component level `CSS` is not optimizing i.e.  it is not loading from the `app.css `instead it is loading from respective components
